randomwallpaper@iflow.space/sourceAdapter.js

```diff
@@ -204,50 +205,32 @@ var WallhavenAdapter = class extends BaseAdapter {
 
 		this._readOptionsFromSettings();
 		let optionsString = this._generateOptionsString();
-		let url = 'https://wallhaven.cc/search?' + encodeURI(optionsString);
+		let url = 'https://wallhaven.cc/api/v1/search?' + encodeURI(optionsString);
 
 		let message = Soup.Message.new('GET', url);
 
 		session.queue_message(message, (session, message) => {
-			let body = message.response_body.data;
-			let urlArray = body.match(new RegExp(/https:\/\/wallhaven.cc\/w\/[0-9a-z]+/g));
+			let images = JSON.parse(message.response_body.data).data;
 
-			if (!urlArray || urlArray.length === 0) {
-				this._error("No image found.", callback);
+			if (!images || images.length === 0) {
+				this._error("No image found. body:"+images, callback);
 				return;
 			}
 
-			// remove dublicates from array
-			let uniqueUrlArray = urlArray.filter(function (item, pos) {
-				return urlArray.indexOf(item) == pos;
-			});
-
 			// get a random entry from the array
-			let url = uniqueUrlArray[Math.floor(Math.random() * uniqueUrlArray.length)];
-
-			message = Soup.Message.new('GET', url);
-
-			if (message === null) {
-				this._error("Could not create request.", callback);
-				return;
-			}
-
-			session.queue_message(message, () => {
-				try {
-					let body = message.response_body.data;
-					let imageDownloadUrl = body.match(new RegExp(/"(https:\/\/w.wallhaven.cc\/full\/.*?)"/))[1];
-
-					if (callback) {
-						let historyEntry = new HistoryModule.HistoryEntry(null, 'Wallhaven', imageDownloadUrl);
-						historyEntry.source.sourceUrl = 'https://wallhaven.cc/';
-						historyEntry.source.imageLinkUrl = url;
-						callback(historyEntry);
-					}
-				} catch (e) {
-					this._error("Unexpected response. (" + e + ")", callback);
-				}
-			})
-
+			let imageUrl = images[Math.floor(Math.random() * images.length)].path;
+
+			let apiKey = this.options["apikey"];
+			if(apiKey){
+                imageUrl += "?apikey="+apiKey;
+            }
+
+            if (callback) {
+                let historyEntry = new HistoryModule.HistoryEntry(null, 'Wallhaven', imageUrl);
+                historyEntry.source.sourceUrl = 'https://wallhaven.cc/';
+                historyEntry.source.imageLinkUrl = url;
+                callback(historyEntry);
+            }
```

Comment:
The indentation here is a little bit off.

Reply:
honesty, I don't see any indentation mistake in here. could you please explain how you want this to be changed?

Reply:
You used spaces for the indentation while the rest of the file uses tabs. But that is not really a problem for me. I'll simply reformat the code on the target branch. Thanks!